### ISSUE TITLE
Stremio Android mobile app doesnt get movies/series information (metadata)

### ISSUE THREAD

**Issue by batuhankrmn** [Stremio/stremio-bugs#828]
**Describe the bug**
When movie or TV show poster is clicked on, content information and episodes didn't appeared on the screen sometimes but if you go back and click on the same poster again, informations and episodes will be showed successfully. Also sometimes informations appears for a second (like flickering) then disappears.

**To Reproduce**
1. Go to discover tab
2. Click on any random movie poster
3. Check movie informations such as description,  genre, imdb rating, cast and also check episodes (If it's TV show)
4. Go back to discover tab again
5. Click on same movie poster again.
6. Check informations from left section.

**Expected behavior**
It's expected that movie/TV shows informations/episodes should be showed successfully when user clicked on the content.

**Screenshots**

https://github.com/Stremio/stremio-bugs/assets/57859656/7ad86f6f-505e-4fcf-b234-6c65f4571f48

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy Tab S9 Ultra
 - OS: Android 14, OneUI 6.1
 - Version 1.6.12 Android App installed from Play Store

Additional Notes:

I've tried clearing cache, force stopping app, reinstalling app but none of them are worked. I believe this is critical issue in terms of user experience because user needs to click twice for every content they want to browse. Hope you can reproduce it on your end and analyze the issue. Best regards.

**Comment by TRtomasz** [comment]
does it happen only from the library?

**Comment by batuhankrmn** [comment]
> does it happen only from the library?

No, It's the same for all pages; Home, Discovery, Library. It's strange issue, seems like it's not related with internet connection or latency because once I go back and click the poster again, metadata comes instantly.

**Comment by batuhankrmn** [comment]
This is from discovery page. Same is valid for all pages. Also please notice that in the end of this screen recording, Mad Max Fury Road had movie description with first click, but missing cast, director, genre etc. After I go back and click to the poster again, movie description is changed and other missing metada is fetched. So I don't know it seemed like something is conflicting to get metadata? 

https://github.com/Stremio/stremio-bugs/assets/57859656/0a4fa232-85aa-439d-ab45-52b0f731e8c3

**Comment by jaruba** [comment]
works fine for me, did u try changing the device DNS to google or cloudflare DNS?

**Comment by batuhankrmn** [comment]
> works fine for me, did u try changing the device DNS to google or cloudflare DNS?

Hey, so I tried changing my DNS to both Google and Cloudflare, but it didn't work. I think we should take a closer look at that last screen recording I sent. Sometimes, when a movie only has a description, I click on the poster again and all the missing information shows up, but the previous description changes too. Doesn't that suggest that the metadata source is changing somehow?

**Comment by jaruba** [comment]
what it suggests is that it's not retrieving the metadata at all for you the first time. it's MetaPreview vs Meta, the catalog items are MetaPreview, when you open a page, you retrieve the Meta (with extended data) from the addon with a different request. until the Meta is retrieved, the app uses the MetaPreview data

so for you, it looks like the first Meta request is either not started at all or failing, but maybe it's just slow for you? and the second time it works because it was cached locally in the meanwhile?

**Comment by batuhankrmn** [comment]
Yeah, I get it now, thanks for explaining. Your theory makes total sense, it could be that, but I have a pretty fast internet connection with low latency (500 megabits) and a powerful device. I'm not sure what else could be causing this slowness. Also, I'm wondering, if the request was slow the first time and then cached and showed up instantly the second time, why can't that process be done on the first request? Like, just wait for the response and then refresh the MetaPreview with the Meta? Actually that's just a suggestion and question, I'm not familiar with the code structure and workflow

**Comment by jaruba** [comment]
well, it should be doing that already, and i'm still uncertain if the request is slow, fails or is never started, but it's strange that others haven't reported this too, which makes it an isolated issue, these are normally caused by the user's environment, but if it's not caused by the DNS, then I can't think of other pausible causes

**Comment by batuhankrmn** [comment]
Actually I made a post on Reddit about this issue and at least 5 people had the same problem. Here's the link: [https://www.reddit.com/r/StremioAddons/s/OtIaD3Ttzt](url)

It might not be happening for everyone, but I think there are others who have this issue and since there's a workaround (second click way), they didn't bother to report it. I want to help you analyze and find the root cause of this. Is there anything else I can share with you, like logs or something?

**Comment by jaruba** [comment]
you should ask the others what country they are in and what ISP they use, we could look over the code and see if we get any ideas, but normally things can't be fixed without being able to reproduce them first

**Comment by keinflausaus** [comment]
I think this is the same issue: #932 

I have also problems with this on the android app

**Comment by TRtomasz** [comment]
closing that one as we are working on new mobile app, that is already available in beta version
